p2p/src/peers.rs

```diff
@@ -518,14 +545,9 @@ impl Peers {
 		}
 	}
 
-	pub fn enough_peers(&self) -> bool {
-		self.peer_count() >= self.config.peer_min_preferred_count()
-	}
-
 	/// We have enough peers, both total connected and outbound connected
 	pub fn healthy_peers_mix(&self) -> bool {
```

Comment:
I'd advise changing the name of this now, since "healthy_peers_mix" doesn't really represent what it's checking. Possible options are something like `meets_preferred_outbound`, or preferably reverse the logic and call it `needs_outbound_connections`. The logic for managing connections was already built piecemeal through a series of defect corrections, so it's probably best to try to keep it from getting anymore complicated/confusing.

Reply:
The naming here wasn't immediately intuitive to me either. My understanding is that it is a result of needing to maintain a healthy mix of outbound connections for dandelion to allow each peer to select a random outbound relay to send txs along the stem. Without ensuring a "healthy peers mix" apparently some nodes were not having enough slots for outbound connections.

With that context it makes a little more sense to me- a healthy peers mix is one
that has at least the minimum number of preferred outbound connections. So while I agree it doesn't strictly represent what is checked (outbound connections >= configured min preferred outbound connections), I think for now it does an ok job of capturing what we are checking for (that we have a healthy peers mix of outgoing connections).

Reply:
Yea, I agree that's where the name came from, but you've now just changed the functionality, so my claim is the name no longer makes sense. Before, it used to compare inbound and outbound count so it made sense, but it's no longer about having a healthy mix, and is now just about making sure we have a minimum number of outbound.

Reply:
Agreed, it could make sense to using a more descriptive name about functionality rather than a prescriptive name about health.